### DISCUSSION TITLE
Data Beeing deleted in the list after changing screens

### DISCUSSION THREAD

**Discussion by harguilar** [FilledStacks/stacked#981] [Q&A]
![image](https://github.com/Stacked-Org/stacked/assets/30771959/52ce1b29-2acf-41e5-8aad-9a77fe1eb7ed)
Hi everyone, 
Need some help with something, 
Am to select 11 player one at any given time and place into a List of player.
I have screen like the one above where you pick 1 player at certain position and take you to another screen where you select from a list of players. 
The problem is that every time I select a new player from a different screen, the previous one get deleted. Meaning am only to put in a field one player. 
I wonder How can I solve this by incrementing the numbers of players without losing  previous ones.

**Comment by ferrarafer** [comment]
Hi how are you doing? I'd like to help but without being able to see the code I'd probably need more wizard skills than developer skills 😄 .

Without knowing exactly how you're managing that, I can give you the advice to use a bottom sheet to display the players in the list so it stays in the same view.

For your implementation, probably the view is being disposed when you navigate to the player list, disable dispose for that view.

**Reply by harguilar** [reply]
@ferrarafer even without being a wizard you still manage to give me a wizard genius idea sorry for stupidity. Below you have the Code of this view as well the viewmodel. I have enable the viewmodel dispose to false but still the issue remains. 


SquadView.dart

@FormView(
  fields: [
    FormTextField(name: 'team', validator: FormValidators.teamValidator),
  ],
  autoTextFieldValidation: true,
)
class SquadView extends StatelessWidget with $SquadView {
  const SquadView({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return ViewModelBuilder<SquadViewModel>.reactive(
      disposeViewModel: false, 
      fireOnViewModelReadyOnce: true,
      // onViewModelReady: (viewModel) => viewModel.fetchListPlayers(),
      builder: (context, viewModel, child) =>  SafeArea(
      child: Scaffold(
        appBar: AppBar(
          title: const Text("Pick Your Team"),
          centerTitle: true,
        ),
        backgroundColor: Colors.black,
        body: viewModel.hasError
            ? Container(
                color: Colors.red,
                alignment: Alignment.center,
                child: const Text(
                  'An error has occered while running the future',
                  style: TextStyle(color: Colors.white),
                ),
              )
            : Center(
                child: viewModel.isBusy
                    ? const CircularProgressIndicator()
                    : Column(
                        children: [
                          CnscLayoutSettings.verticalSpaceMedium,
                          Expanded(
                            child: CnscTournamentInputField(
                              controller: teamController,
                              placeholder: "Enter Team Name",
                              trailing: const Icon(Icons.close),
                              trailingTapped: () => teamController.clear(),
                              //errorText: viewModel.emailOrNameInputValidationMessage,
                            ),
                          ),
                          Expanded(
                            flex: 8,
                            child: SizedBox(
                              height: 8,
                              child: Container(
                                padding: const EdgeInsets.only(
                                    left: 25.0, right: 25.0),
                                constraints: const BoxConstraints.expand(),
                                decoration: BoxDecoration(
                                  image: DecorationImage(
                                    image: AssetImage(IMAGES.ftpitch),
                                    fit: BoxFit.cover,
                                  ),
                                ),
                                child: Column(
                                  children: [
                                    /*         const CnscTournamentText(
                                          text: 'GOAL-KEEPERS',
                                          //style: Get.theme.textTheme.headline6,
                                        ), */
                                    Padding(
                                      padding: const EdgeInsets.only(top: 15),
                                      child: Row(
                                        mainAxisAlignment:
                                            MainAxisAlignment.spaceEvenly,
                                        children: [
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 0);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 50),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child:
                                                                CnscTournamentText(
                                                              text: viewModel
                                                                  .data![0].name
                                                                  .toString(),
                                                              //style: Get.theme.textTheme.bodyText2,  '',
                                                            ),
                                                          ),
                                                        ),
                                                        /*   const CnscTournamentText(
                                            text: '9.5 cr',
                                            //style: Get.theme.textTheme.headline6,
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                        ],
                                      ),
                                    ),
                                    Padding(
                                      padding: const EdgeInsets.only(top: 10),
                                      child: Row(
                                        mainAxisAlignment:
                                            MainAxisAlignment.spaceEvenly,
                                        children: [
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 1);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child:
                                                                CnscTournamentText(
                                                              text: viewModel
                                                                  .data![1].name
                                                                  .toString(),
                                                            ),
                                                          ),
                                                        ),
                                                        /*   const CnscTournamentText(
                                            text: '8.5 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 2);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child:
                                                                CnscTournamentText(
                                                              text: viewModel
                                                                  .data![2].name
                                                                  .toString(),
                                                            ),
                                                          ),
                                                        ),
                                                        /*    const CnscTournamentText(
                                            text: '8.5 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 3);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child: CnscTournamentText(
                                                                text: viewModel
                                                                    .data![3]
                                                                    .name
                                                                    .toString()),
                                                          ),
                                                        ),
                                                        /*      const CnscTournamentText(
                                            text: '9 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 4);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(Radius
                                                                        .circular(
                                                                            5)),
                                                          ),
                                                          child: Center(
                                                            child: CnscTournamentText(
                                                                text: viewModel
                                                                    .data![4]
                                                                    .name
                                                                    .toString()),
                                                          ),
                                                        ),
                                                        /*    const CnscTournamentText(
                                            text: '9.5 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                        ],
                                      ),
                                    ),
                                    Padding(
                                      padding: const EdgeInsets.only(top: 95),
                                      child: Row(
                                        mainAxisAlignment:
                                            MainAxisAlignment.spaceEvenly,
                                        children: [
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 5);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child:
                                                                CnscTournamentText(
                                                              text: viewModel
                                                                  .data![5].name
                                                                  .toString(),
                                                            ),
                                                          ),
                                                        ),
                                                        /*    const CnscTournamentText(
                                            text: '9 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 6);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child:
                                                                CnscTournamentText(
                                                              text: viewModel
                                                                  .data![6].name
                                                                  .toString(),
                                                            ),
                                                          ),
                                                        ),
                                                        /*     const CnscTournamentText(
                                            text: '8.5 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 7);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 55),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child: CnscTournamentText(
                                                                text: viewModel
                                                                    .data![7]
                                                                    .name
                                                                    .toString()),
                                                          ),
                                                        ),
                                                        /*      const CnscTournamentText(
                                            text: '9.5 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                        ],
                                      ),
                                    ),
                                    Padding(
                                      padding: const EdgeInsets.only(top: 90),
                                      child: Row(
                                        mainAxisAlignment:
                                            MainAxisAlignment.spaceEvenly,
                                        children: [
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 8);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(Radius
                                                                        .circular(
                                                                            5)),
                                                          ),
                                                          child: Center(
                                                            child: CnscTournamentText(
                                                                text: viewModel
                                                                    .data![8]
                                                                    .name
                                                                    .toString()),
                                                          ),
                                                        ),
                                                        /*    const CnscTournamentText(
                                            text: '9 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 9);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child: CnscTournamentText(
                                                                text: viewModel
                                                                    .data![9]
                                                                    .name
                                                                    .toString()),
                                                          ),
                                                        ),
                                                        /*      const CnscTournamentText(
                                            text: '9.5 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                          SizedBox(
                                            child: GestureDetector(
                                              onTap: () {
                                                viewModel.playerInPitch(
                                                    playerTracker: 10);
                                              },
                                              child: Stack(
                                                children: [
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            left: 20),
                                                    child: Container(
                                                      height: 60,
                                                      width: 40,
                                                      decoration: BoxDecoration(
                                                        image: DecorationImage(
                                                          image: AssetImage(
                                                              IMAGES.pShirt),
                                                          fit: BoxFit.cover,
                                                        ),
                                                      ),
                                                    ),
                                                  ),
                                                  Padding(
                                                    padding:
                                                        const EdgeInsets.only(
                                                            top: 45),
                                                    child: Column(
                                                      children: [
                                                        Container(
                                                          height: 20,
                                                          width: 80,
                                                          decoration:
                                                              const BoxDecoration(
                                                            color: Colors.white,
                                                            borderRadius:
                                                                BorderRadius
                                                                    .all(
                                                              Radius.circular(
                                                                  5),
                                                            ),
                                                          ),
                                                          child: Center(
                                                            child: CnscTournamentText(
                                                                text: viewModel
                                                                    .data![10]
                                                                    .name
                                                                    .toString()),
                                                          ),
                                                        ),
                                                        /*       const CnscTournamentText(
                                            text: '9 cr',
                                          ), */
                                                      ],
                                                    ),
                                                  ),
                                                ],
                                              ),
                                            ),
                                          ),
                                        ],
                                      ),
                                    ),
                                  ],
                                ),
                              ),
                            ),
                          ),
                        ],
                      ),
              ),
        floatingActionButton: Positioned(
          child: FloatingActionButton.extended(
            onPressed: () {
              viewModel.insertRecords(); 
              
              // final text = teamController.text.toString();

              //print ('Harguilar This is the valeu {$text}');
            },
            icon: const Icon(Icons.save_sharp),
            label: const Text('Save Team'),
            backgroundColor: Colors.black12,
          ),
        ),
      ),
    ),
  
      viewModelBuilder: () => SquadViewModel(),
    );
  }
}

SquadViewModel.dart. 

class SquadViewModel extends FutureViewModel<List<Player>?> with FormStateHelper{
  final _navigationService = locator<NavigationService>();
  final _footBallService = locator<FootballService>();
  final _userService = locator<UserService>();
  final _localStorageService = locator<LocalsecurestorageService>();
  final firestore_api = locator<FirestoreapiService>(); 
  final log = getLogger('SquadViewModel');



  User? _currentUser;

  int? playerPosition;
  bool validator = false;

  List<Player>? listPlayers = [];
  Player? player;
  var dataInformation; 
  var dataInfo; 
  final String players = "player"; 


  //Set List of Players
  Future<List<Player>?> getPlayerListAndPosition() async {
    
    player = _footBallService.getPlayer;

    playerPosition = _footBallService.getPlayerPositionFromSquad;

    _currentUser = _userService.getCurrentUser;

    

    validator = player?.name.isNotEmpty ?? false; 

    if (validator){

    

      dataInformation!.removeAt(playerPosition!); 

      dataInformation!.insert(playerPosition!, player!); 

 

      //notifyListeners(); 
      return  dataInformation!; 
    }   
     dataInformation =  await _footBallService.getListDefaultPlayers(user: _currentUser!);
      
    
    
      return dataInformation;    
  }


  void playerInPitch({required int playerTracker}) {
    log.i("I just Clicked Now");

    /// Set the Player Current
    _footBallService.setPlayerPosFromSquad(
        playerPositionTracker: playerTracker);

    //Navegate to Select Players
    _navigationService.navigateTo(Routes.footballLineupView);
  }

  @override
  void onError(error) {}

  @override
  Future<List<Player>?> futureToRun()=> getPlayerListAndPosition();
    
}

**Comment by ferrarafer** [comment]
@harguilar were you able to fix your issue?

**Reply by harguilar** [reply]
Hi @ferrarafer, 
Thanks for getting back to me. To be honest I did not manage to resolve the issue. 
I decided to use the Bottom Sheet as you advised but am running into the issue that when I select a player from the Bottom Sheet View, and I want to pass it Squad view it does not show up even though I called notify listeners in the squadview to refresh my UI.  Please see the code below. 

Bottom Sheet Model 
class SelectPlayersSheetModel extends FutureViewModel <List<Player>?> with FormStateHelper {
  //final _bottomSheetService = locator<BottomSheetService>();
  bool? _confirmationResult;
  bool? get confirmationResult => _confirmationResult;
  final log = getLogger('SelectPlayersSheetModel'); 
  final _navigationService = locator<NavigationService>();  


  final _playersServices = locator<PlayersServicesService>(); 

  //final _navigationService = locator<NavigationService>(); 

  final _footBallService = locator<FootballService>(); 


   void setPlayerPositionFootBall({required Player player, required bool validateSelectedPlayer}){

    _footBallService.setPlayerPositionFootBall(player: player, validator: validateSelectedPlayer);


    log.i("This is the player I clicked {$player}");

    _navigationService.replaceWithSquadView(); 

  }

    //Set List of Players 
   Future<List<Player>?> getPlayersList () async {
  
      return await _playersServices.getListPlayers();    
  } 

  @override
  Future <List<Player>?> futureToRun() => getPlayersList (); 
}

Squad Model 

class SquadViewModel extends FutureViewModel<List<Player>?>
    with FormStateHelper {
  // final _navigationService = locator<NavigationService>();
  final _bottomSheetService = locator<BottomSheetService>();
  final _footBallService = locator<FootballService>();
  final _userService = locator<UserService>();
  //final _localStorageService = locator<LocalsecurestorageService>();
  //final firestore_api = locator<FirestoreapiService>();

  final log = getLogger('SquadViewModel');
  User? _currentUser;
  int? playerPosition;
  bool validateSelectedPlayer = false;
  List<Player>? listPlayers = [];
  Player? player;
  // var dataInformation;
  //var dataInfo;
  final String players = "player";

  //Set List of Players
  Future<List<Player>?> getPlayerListAndPosition() async {

    _currentUser = _userService.getCurrentUser;

    validateSelectedPlayer = _footBallService.getValidatedSelectedPlayer; 

    // final prefs = await SharedPreferences.getInstance();

    if (validateSelectedPlayer) {
      try {
        player = _footBallService.getPlayer;

        playerPosition = _footBallService.getPlayerPositionFromSquad;
        //QuerySnapshot query;
        //log.i('Harguilar THis is the Value inside {$dataInformation}');
        log.i('Harguilar THis is the Value inside {$listPlayers}');
        if (listPlayers!.isNotEmpty){
          listPlayers!.removeAt(playerPosition!);
          listPlayers!.insert(playerPosition!, player!);
        }
        else {
          listPlayers!.insert(playerPosition!, player!);

        }
        
        notifyListeners();

        return listPlayers!;
      } catch (e) {
        log.i(' This is my Error ${e.toString()}');
      }
    }

    listPlayers =
        await _footBallService.getListDefaultPlayers(user: _currentUser!);

    return listPlayers;
  }

  Future<void> playerInPitch({required int playerTracker}) async {
    log.i("I just Clicked Now");

    /// Set the Player Current
    _footBallService.setPlayerPosFromSquad(
        playerPositionTracker: playerTracker);

    //Navegate to Select Players
    //_navigationService.navigateTo(Routes.footballLineupView);
    //  _navigationService.clearStackAndShow(Routes.footballLineupView);

    await _bottomSheetService.showCustomSheet(
        variant: BottomSheetType.selectPlayers, title: "Select A Player");
  }

  /*  Future<void> insertRecords() async {
    //await firestore_api.exportData();
    await firestore_api.updloadFile();
  }
 */
  @override
  void onError(error) {}

  @override
  Future<List<Player>?> futureToRun() => getPlayerListAndPosition();
}

**Comment by ferrarafer** [comment]
I didn't have time to take a look to this, sorry. Did you resolve it?
